pkg/api/routes.go

```diff
@@ -149,5 +155,11 @@ func (h *Handler) GetRoutes() []Route {
 			"/log",
 			h.SetLogLevel,
 		},
+		{
+			"GetMetrics",
+			"GET",
+			"/metrics",
```

Comment:
Yes, Prometheus will periodically hit this route and the `promhttp.Handler` will handle the request.  He will then call the `Collect()` function on the ceph exporter, will returns everything to the Prometheus handler via channels.  Then the prometheus handler writes the http response with all the metrics we returned to it.

The default refresh/scrape interval is I believe every 20 or 30 seconds.  Even if it were faster, I don't imaging too much of a performance hit from this because ceph is doing the book keeping for all these metrics whether or not we request them.  We just make a handful of mon commands over the cluster network to get the metrics from ceph and I am not under the impression that is very expensive.